cql3/statements/select_statement.cc

```diff
@@ -1118,12 +1118,15 @@ query::partition_slice indexed_table_select_statement::get_partition_slice_for_g
     partition_slice_builder partition_slice_builder{*_view_schema};
 
     if (!_restrictions->has_partition_key_unrestricted_components()) {
+        // Computed token column to be added to index view restrictions
+        const column_definition& token_cdef = *_view_schema->clustering_key_columns().begin();
+        // Is secondary index with new token_column_computation type (computation name: token_v2)
+        const bool is_token_v2_index = token_cdef.type == long_type;
+
         auto single_pk_restrictions = dynamic_pointer_cast<restrictions::single_column_partition_key_restrictions>(_restrictions->get_partition_key_restrictions());
-        // Only EQ restrictions on base partition key can be used in an index view query
+        // Only EQ restrictions or TOKEN restriction on base partition key can be used in an index view query
```

Comment:
Confusing: TOKEN will make single_pk_restrictions null.

Reply:
`single_pk_restrictions` is only used in `EQ` case. In the amended commit, I split this comment into two, so hopefully it is not as confusing.